<?xml version="1.0" ?>
<launch>
 
 <arg name="motion-1"   default="false"/>
 <arg name="motion-2"   default="true"/>

# input_topic name should contatin: geometry_msgs::Pose
# output_topic_name will write geometry_msgs::Twist
<group if="$(arg motion-1)">
  <group ns="ds1">
    <param name="input_topic_name" value="/lwr/ee_pose"/>
    <param name="output_topic_name" value="/ds1/desired_velocity"/>
    <param name="output_filtered_topic_name" value="/lwr/joint_controllers/passive_ds_command_vel"/>
    <rosparam command="load" file="$(find ds_motion_generator)/config/free_ws.yml"/>
    <node name="ds_node" pkg="ds_motion_generator" type="active_motionGenerator_node" output="screen"/>
  </group>
</group>

 <!-- Startup RVIZ with given config -->
<group if="$(arg motion-2)">
  <group ns="ds1">
    <param name="input_topic_name" value="/lwr/ee_pose"/>
    <param name="output_topic_name" value="/ds1/desired_velocity"/>
    <param name="output_filtered_topic_name" value="/lwr/joint_controllers/passive_ds_command_vel"/>
    <rosparam command="load" file="$(find ds_motion_generator)/config/Curve_go_right.yml"/>
<node name="ds_node" pkg="ds_motion_generator" type="active_motionGenerator_node" output="screen"/>
  </group>
</group>

</launch>
